apis/cloud.redhat.com/v1alpha1/clowdapp_types.go

```diff
@@ -189,6 +189,16 @@ type Deployment struct {
 
 	// AutoScaler defines the configuration for the auto scaler
 	AutoScaler *AutoScaler `json:"autoScaler,omitempty"`
+
+	// DeploymentStrategy allows the deployment strategy to be set only if the
+	// deployment has no public service enabled
+	DeploymentStrategy *DeploymentStrategy `json:"deploymentStrategy,omitempty"`
+}
+
+type DeploymentStrategy struct {
+	// OverridePrivate allows a deployment that only uses a private port to use
+	// the Rolling update, as the default is Recreate.
+	OverridePrivate bool `json:"overridePrivate,omitempty"`
```

Comment:
why not just name it `Strategy` or whatever mirrors the actual `DeploymentStrategy` struct?

Reply:
Because if you see - we are not giving the full struct here - we decided to just go with the bool right?

Reply:
done